utils/apiFeatures.js

```diff
@@ -0,0 +1,26 @@
+class APIFeatures {
```

Comment:
This file and the ones in the `questions` folder should be the only ones in this commit as they introduce pagination. We can treat the password update changes in it's own PR.